aci/resource_aci_cloudtemplatevpnnetwork.go

```diff
@@ -0,0 +1,498 @@
+
+		cloudIpSecTunnelAttMap, cloudtemplateIpSecTunnelDn, err := setCloudTemplateforIpSecTunnelAttributes(cloudtemplateIpSecTunnel, make(map[string]interface{}))
+		if err != nil {
+			d.SetId("")
+			return nil, err
+		}
+
+		log.Printf("[DEBUG] Begining Import of cloud BGP IPV4 Peer attributes.")
+		bgpIPv4PeerData, err := aciClient.ListCloudTemplateBGPIPv4Peer(cloudtemplateIpSecTunnelDn)
+		if err != nil {
+			log.Printf("[DEBUG] Error while importing cloud BGP IPV4 Peer attributes %v", err)
+		}
+		for _, bgpIPv4Peer := range bgpIPv4PeerData {
+			bgpPeerAsnAtt, err := getASNfromBGPTPV4Peer(bgpIPv4Peer, make(map[string]string))
+			if err != nil {
+				d.SetId("")
+				return nil, err
+			}
+			cloudIpSecTunnelAttMap["bgp_peer_asn"] = bgpPeerAsnAtt["bgp_peer_asn_att"]
+		}
+		log.Printf("[DEBUG] Import of cloud BGP IPV4 Peer finished successfully.")
+
+		cloudtemplateIpSecTunnelSet = append(cloudtemplateIpSecTunnelSet, cloudIpSecTunnelAttMap)
+	}
+	d.Set("ipsec_tunnel", cloudtemplateIpSecTunnelSet)
+	log.Printf("[DEBUG] Import of cloud IPSec Tunnel finished successfully.")
+
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciCloudTemplateforVPNNetworkCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] TemplateforVPNNetwork: Beginning Creation")
+	aciClient := m.(*client.Client)
+	name := d.Get("name").(string)
+	TemplateforExternalNetworkDn := d.Get("aci_cloud_external_network_dn").(string)
+
+	cloudtemplateVpnNetworkAttr := models.CloudTemplateforVPNNetworkAttributes{}
+
+	nameAlias := ""
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		nameAlias = NameAlias.(string)
+	}
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		cloudtemplateVpnNetworkAttr.Annotation = Annotation.(string)
+	} else {
+		cloudtemplateVpnNetworkAttr.Annotation = "{}"
+	}
+
+	if Name, ok := d.GetOk("name"); ok {
+		cloudtemplateVpnNetworkAttr.Name = Name.(string)
+	}
+
+	if RemoteSiteId, ok := d.GetOk("remote_site_id"); ok {
+		cloudtemplateVpnNetworkAttr.RemoteSiteId = RemoteSiteId.(string)
+	}
+
+	if RemoteSiteName, ok := d.GetOk("remote_site_name"); ok {
+		cloudtemplateVpnNetworkAttr.RemoteSiteName = RemoteSiteName.(string)
+	}
+	cloudtemplateVpnNetwork := models.NewCloudTemplateforVPNNetwork(fmt.Sprintf(models.RncloudtemplateVpnNetwork, name), TemplateforExternalNetworkDn, nameAlias, cloudtemplateVpnNetworkAttr)
+
+	err := aciClient.Save(cloudtemplateVpnNetwork)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if ipSecTunnelPeers, ok := d.GetOk("ipsec_tunnel"); ok {
+		clopudIpSecTunnels := ipSecTunnelPeers.(*schema.Set).List()
+		// Looping throught List of IPSec Tunnels
+		for _, val := range clopudIpSecTunnels {
+			ipSecTunnels := val.(map[string]interface{})
+
+			cloudtemplateIpSecTunnelAttr := models.CloudTemplateforIpSectunnelAttributes{}
+			cloudtemplateIpSecTunnelAttr.Annotation = "{}"
```

Comment:
do we need annotation t match parent annotation or do define annotation in child? we always set to fixed value now